apps/api/v2/src/modules/users/inputs/create-user.input.ts

```diff
@@ -1,6 +1,28 @@
-import { IsString } from "class-validator";
+import { IsTimeFormat } from "@/modules/users/inputs/validators/is-time-format";
+import { IsTimeZone } from "@/modules/users/inputs/validators/is-time-zone";
+import { IsWeekStart } from "@/modules/users/inputs/validators/is-week-start";
+import { IsNumber, IsOptional, IsString, Validate } from "class-validator";
 
 export class CreateUserInput {
   @IsString()
   email!: string;
+
+  @IsNumber()
+  @IsOptional()
+  @Validate(IsTimeFormat)
+  timeFormat?: number;
+
+  @IsNumber()
+  @IsOptional()
+  defaultScheduleId?: number;
```

Comment:
We need a default schedule id at the time of creating a user in order for our default schedule endpoint to work so I though it's better if `defaultScheduleId` isn't optional for the create user input

Reply:
The thing is that when creating a schedule `userId` is needed if `schedule.availabilities` are passed in the request body - see function calls within `createSchedule` of `apps/api/v2/src/ee/schedules/controllers/schedules.controller.ts`, so there might be a situation where we can't make a schedule without having a user created before.

So we can either:
1. Add new optional property `isDefault` that is "true" by default when creating a schedule, and without default value when updating it.
2. Create a new schedules endpoint POST /default that accepts request body { scheduleId: number } and sets "defaultScheduleId" of user to it.
3. Don't do anything and tell our users to create user, create schedule, and then make call to update user with the defaultScheduleId of the schedule.

I went with option 1 for now in [refactor: default schedule in schedules endpoints](https://github.com/calcom/cal.com/pull/13486/commits/eb0dfc4fa5bb222e2db70f311e843de5d4e46c56)

Reply:
So now after creating a user schedule it will be sure that the user has `defaultScheduleId` set.